ServerCore/PuzzleStateHelper.cs

```diff
@@ -193,8 +208,90 @@ public static async Task SetSolveStateAsync(PuzzleServerContext context, Event e
             // if this puzzle got solved, look for others to unlock
             if (value != null)
             {
-                await UnlockAnyPuzzlesThatThisSolveUnlockedAsync(context, eventObj, puzzle, team, value.Value);
+                await UnlockAnyPuzzlesThatThisSolveUnlockedAsync(context,
+                    eventObj,
+                    puzzle,
+                    team,
+                    value.Value);
+            }
+        }
+
+        /// <summary>
+        /// Set the email only mode of some puzzle state records. In the course
+        /// of setting the state, instantiate any state records that are
+        /// missing on the server.
+        /// </summary>
+        /// <param name="context">The puzzle DB context</param>
+        /// <param name="eventObj">The event we are querying from</param>
+        /// <param name="puzzle">
+        ///     The puzzle; if null, get all puzzles in the event.
+        /// </param>
+        /// <param name="team">
+        ///     The team; if null, get all the teams in the event.
+        /// </param>
+        /// <param name="value">The new email only state for the puzzle</param>
+        /// <param name="author"></param>
+        /// <returns>
+        ///     A task that can be awaited for the lockout operation
+        /// </returns>
+        public static async Task SetEmailOnlyModeAsync(
```

Comment:
We should have a "teams in email mode" page that authors/admins can use to kick teams out. Doesn't have to be part of this PR but let's not lose sight of it.